src/Interpreters/OptimizeIfWithConstantConditionVisitor.cpp

```diff
@@ -49,6 +49,16 @@ static bool tryExtractConstValueFromCondition(const ASTPtr & condition, bool & v
                 }
             }
         }
+        else if (function->name == "toUInt8" || function->name == "toInt8")
```

Comment:
Do we still need `OptimizeIfWithConstantCondition`?
Also maybe make a setting to disable this optimization?

Reply:
Looks like we need it, there is no other way to skip unused columns

Reply:
Where exactly it is needed?